src/index.ts

```diff
@@ -657,12 +660,43 @@ export default class StepFunctionsOfflinePlugin implements Plugin {
     return waitTimer;
   }
 
-  createContextObject(states: StateMachine['States'], isSubContext: boolean): ContextObject {
+  createContextObject(
+    states: StateMachine['States'],
+    name: string,
+    originalEvent: Event,
+    isSubContext: boolean
+  ): ContextObject {
+    let attempt = 0;
     const cb = (err: Maybe<Error>, result?: Event) => {
-      if (err) {
+      if (!notEmpty(this.currentState)) return;
+      if (err && !isType('Task')<Task>(this.currentState)) {
         throw `Error in function "${this.currentStateName}": ${JSON.stringify(err)}`;
       }
-      if (!notEmpty(this.currentState)) return;
+      if (err && isType('Task')<Task>(this.currentState)) {
+        const matchingError = (this.currentState.Retry ?? []).find(condition =>
+          condition.ErrorEquals.includes('HandledError')
+        );
+        if (!matchingError) throw `Error in function "${this.currentStateName}": ${JSON.stringify(err)}`;
+        attempt += 1;
+        if (attempt < (matchingError.MaxAttempts ?? 0)) {
```

Comment:
Does this need to be `<=`?

Reply:
attempt starts at 1, so attempt=5 would be execution=6, we actually want to attempt it 5 times=6 executions